docs/CI-PIPELINES.md

```diff
@@ -0,0 +1,420 @@
+* The job dsl definition script (netci.groovy) file tends to become very difficult to decipher over time.  They tend to grow into a spaghetti code mass.
+
+Jenkins Pipelines were introduced a few years back and became a 'first class' citizen in Jenkins with the Jenkins 2.0 overhaul.  They solve many of the issues with freestyle jobs. A pipeline job reads a script from source control and executes it.  This script interacts with the Jenkins CI system to allocate nodes, run commands, gather results, etc.
+
+# .NET CI System Lifecycle - FreeStyle vs. Pipelines
+
+Assuming some familiarity with the existing .NET CI system (netci.groovy), this section details the current FreeStyle (netci.groovy) lifecycle and how the Jenkins pipeline workflow is different.
+
+## Freestyle Jobs Today
+
+1. User adds an entry to the [repo list](../data/repolist.txt).
+2. Each jenkins server notices a change to the [repo list](../data/repolist.txt) and reads the file.  It creates a folder for each entry in the list that is targeted at the server and a 'generator' job which watches the repo/branch combo for the changes in the CI definition file (usually netci.groovy)
+3. When a new CI definition file is committed to the repo, Jenkins runs the netci.groovy file.  This file is effectively a script that sets up state in Jenkins.  In this case, the state consists of the set of jobs, their triggers, views, etc. that are defined for the repo.
+4. The created jobs start waiting for their triggers (PR test, PR comment, etc.)
+5. Upon triggering, Jenkins reads the set of steps preconfigured in the job (source to clone, build steps to run, test results to publish) and executes them on the target machine pool configured in the job.
+
+## The New World Of Pipelines
+
+1. User adds an entry to the [repo list](../data/repolist.txt).
+2. Each jenkins server notices a change to the [repo list](../data/repolist.txt) and reads the file.  It creates a folder for each entry in the list that is targeted at the server and a 'generator' job which watches the repo/branch combo for the changes in a 'pipeline declaration' file (usually pipelines.groovy)
+3. When a new pipeline declaration file is committed to the repo, Jenkins runs it. The pipeline declaration file is similar to the traditional CI definition file, but is more barebones.  It notes the pipeline script to run in the repo, the parameters to pass it, and the triggers on which to run it.  The entire workload logic is contained within the pipeline itself, which is read from source control.
+4. The created pipeline jobs start waiting for their triggers (PR test, PR comment, etc.)
+5. Upon triggering, Jenkins reads the configured pipeline script configured in the pipeline job, typically from source control, passes it the default set of parameters configured in the pipeline declaration file, and executes the pipeline.
```

Comment:
`reads the configured pipeline script configured in the pipeline job` is a bit confusing. Do you mind rephrasing it?